### PR TITLE
Introduce util method for moving rows command

### PR DESCRIPTION
**Pull request by jomarko**
@manstis please have a look

### CODE REVIEW

**Comment by manstis** [comment]
Sorry @jomarko another day, another need to rebase your PRs :-(

**Comment by jomarko** [comment]
@manstis no problem, easy to rebase, please have a look now.

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/commands/util/CommandUtils.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+public class CommandUtils {
+
+    public static <T> void updateRowNumbers(final DMNGridData uiModel,
+                                            final Class<T> rowNumberColumnClass,
+                                            final IntStream rangeOfRowsToUpdate) {
+        final Optional<T> rowNumberColumn = uiModel
+                .getColumns()
+                .stream()
+                .filter(c -> rowNumberColumnClass.isInstance(c))
```
Comment (manstis) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/commands/util/CommandUtils.java]: Have you tried this on the client-side?

AFAIK the only methods emulated on ```Class``` are [these](http://www.gwtproject.org/doc/latest/RefJreEmulation.html#Package_java_lang). I had a quick peek at ```com.google.gwt.emul.java.lang.Class``` in ```gwt-user.jar``` and these methods do not appear either.

Comment (jomarko) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/commands/util/CommandUtils.java]: good catch, amending

**Comment by jomarko** [comment]
@manstis added separate commit

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/commands/util/CommandUtils.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    public static void updateRowNumbers(final DMNGridData uiModel,
+                                        final IntStream rangeOfRowsToUpdate) {
+        final Optional<RowNumberColumn> rowNumberColumn = uiModel
+                .getColumns()
+                .stream()
+                .map(c -> {
+                    if (c instanceof DecisionTableRowNumberColumn) {
+                        return (DecisionTableRowNumberColumn) c;
+                    } else {
+                        return (RowNumberColumn) c;
```
Comment (manstis) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/commands/util/CommandUtils.java]: Won't this cast break on the first non-```RowNumberColumn``` encountered on the model!?

Comment (manstis) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/commands/util/CommandUtils.java]: Of course, ```RowNumberColumn``` is always the first column so the CCE would never occur in practice.. however if that's the case then why not just use ```uiModel.getColumns().get(0)```.... All said and done, did you notice ```DecisionTableRowNumberColumn``` extends ```RowNumberColumn``` so you only really need to check if the column is instanceof ```RowNumberColumn``` and both classes will match. I think that'd be the best option.

Comment (jomarko) [reply kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/commands/util/CommandUtils.java]: @manstis paradox, usually I am asking you for defensive approach, this time we changed roles, sure, will amend :)

**Comment by jomarko** [comment]
@manstis sorry wasting your time, please have another look

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/commands/util/CommandUtils.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+import org.uberfire.ext.wires.core.grids.client.widget.grid.columns.RowNumberColumn;
+
+public class CommandUtils {
+
+    public static void updateRowNumbers(final DMNGridData uiModel,
+                                        final IntStream rangeOfRowsToUpdate) {
+        final Optional<RowNumberColumn> rowNumberColumn = uiModel
+                .getColumns()
+                .stream()
+                .map(column -> {
```
Comment (manstis) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/commands/util/CommandUtils.java]: Why not simply remove the map and replace with a ```.filter(c -> c instanceof RowNumberColumn)```?

**Comment by jomarko** [comment]
@manstis now?

**Comment by jomarko** [comment]
It was a ride, thanks!

**Pull request merged**